### ISSUE TITLE
PatchParams dry-run doesn't seems to work

### ISSUE THREAD

**Issue by shigedangao** [clux/kube-rs#511]
Hello,

First of all I want to thank you all for this wonderful library. I'm pretty new in Rust and I'm currently trying to use the PatchParams in order to use the dry-run feature. I'm currently following this example [Example](https://docs.rs/kube/0.52.0/kube/api/struct.Api.html#method.patch). However I'm encountering an error which you can find below

```
PatchOptions.meta.k8s.io "" is invalid: dryRun: Unsupported value: []string{"true"}: supported values: "All": Invalid (ErrorResponse { status: "Failure", message: "PatchOptions.meta.k8s.io \"\" is invalid: dryRun: Unsupported value: []string{\"true\"}: supported values: \"All\"", reason: "Invalid", code: 422 })
```

Below is the code I'm using for my experiment 

```rust
let patch: Value = serde_json::from_str(&output).unwrap();

let params = PatchParams::apply("kubectl").dry_run();
let patch = Patch::Apply(&patch);

let client = Client::try_default()
    .await
    .map_err(|err| KubeError { message: err.to_string() })?;

let dep: Api<Deployment> = Api::namespaced(client, "default"); 
let res = dep.patch("rusty", &params, &patch)
     .await
     .map_err(|err| KubeError { message: err.to_string() })?;
```

The error seems to indicate that the  ```dryRun``` is sending the value ```true``` instead of ```All```. However I don't really know if it's tight to a kubernetes version or not. What do you think ? (My tests are done on k3d, and GKE v1.17.17-gke.3700)

**Analysis of the issue & trying to resolve the issue**

From what I could found, it seems that the dryRun value is set to ```true```  by the ```populate_qp``` method (located in the **kube/src/api/params.rs**). Locally I edited the library and changed the value from ```true``` to ```All``` and seems to work.

**Comment by clux** [comment]
Oh, that's almost certainly an oversight. We have the correct serialization to do this in `DeleteParams` already (in `params.rs`), but it's not ported to `PatchParams`. Thanks for the report, this should be an easy fix!

**Comment by clux** [comment]
Pr in #511, untested but think it should work. Will verify tomorrow.

**Comment by shigedangao** [comment]
@clux seems good to me ! 💯

**Comment by clux** [comment]
Released in 0.53.0. Thanks again.
